src/utils.ts

```diff
@@ -233,3 +235,45 @@ export class RKIError extends Error {
     this.url = url;
   }
 }
+
+export function parseDate(dateString: string): Date {
+  const parts = dateString.split(",");
+  const dateParts = parts[0].split(".");
+  const timeParts = parts[1].replace("Uhr", "").split(":");
+  return new Date(
+    parseInt(dateParts[2]),
+    parseInt(dateParts[1]) - 1,
+    parseInt(dateParts[0]),
+    parseInt(timeParts[0]),
+    parseInt(timeParts[1])
+  );
+}
+
+export async function getDataAlternateSource(url: string, blId?: string) {
+  // If a specific table is given download this state data only
+  let stateIdList = [];
+  for (let id = 1; id <= 16; id++) {
+    stateIdList[id - 1] = id.toString().padStart(2, "0");
+  }
+  if (blId && stateIdList.includes(blId)) {
+    const urlOne = url.replace("Covid19_hubv", `Covid19_${blId}_hubv`);
+    const response = await axios.get(urlOne);
+    var data = response.data;
+  }
+  // else download all 16 state data
+  else {
+    const urlOne = url.replace("Covid19_hubv", "Covid19_01_hubv");
+    let response = await axios.get(urlOne);
+    var data = response.data;
+    for (let i = 2; i <= 16; i++) {
+      const id = i.toString().padStart(2, "0");
+      const urlX = url.replace("Covid19_hubv", `Covid19_${id}_hubv`);
+      response = await axios.get(urlX);
```

Comment:
We should not use `await` in a loop, this causes the get requests to wait for one request to finish before starting the next. It is faster to collect the promises in an array and call `Promise.all` (https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/all)

Reply:
also done